### PR TITLE
Added RenderGeometry handling and RenderMesh method

### LINKED ISSUE #105: Querying RenderMesh of a RenderGeometry causes a Stack Overflow.
#### Description:
The `RenderMesh(obj)` method in Graphics_Engine is called recursively as a result of the `IRenderMesh` method not handling `RenderGeometry`. 

#### Steps to reproduce:
Plug a `RenderGeometry` into `Query.RenderMesh`

#### Expected behaviour:
`RenderMesh` of `RenderGeometry` returned with color and geometry as appropriate.

### PR DESCRIPTION
**Pull request by enarhi**
### Issues addressed by this PR

Closes #105 

Added handling and method for getting RenderMesh from RenderGeometry.

### Test files
[test file](https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?newTargetListUrl=%2Fsites%2FBHoM%2F02%5FCurrent&viewpath=%2Fsites%2FBHoM%2F02%5FCurrent%2FForms%2FAllItems%2Easpx&id=%2Fsites%2FBHoM%2F02%5FCurrent%2F12%5FScripts%2F02%5FPull%20Request%2FBHoM%2FTriangleNet%5FToolkit%2F%23107%2DRenderMeshFromRenderGeometry&viewid=a5a93cba%2Dfcca%2D46cc%2Da31a%2D80b8d5cbfd7b)

### CODE REVIEW

File: TriangleNet_Engine/Compute/Representation/RenderMesh/Other/RenderGeometry.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
```
Comment (FraserGreenroyd) [inline TriangleNet_Engine/Compute/Representation/RenderMesh/Other/RenderGeometry.cs]:
```suggestion
 * Copyright (c) 2015 - 2023, the respective contributors. All rights reserved.
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
Will leave functionality review to @alelom - but the issue it's closing doesn't seem to exist? I'm thinking it should be #105 @enarhi but could you confirm and update as necessary?

File: TriangleNet_Engine/Compute/Representation/RenderMesh/Other/RenderGeometry.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        /***************************************************/
+        /**** Public Methods - Graphics                 ****/
+        /***************************************************/
+
+        public static BH.oM.Graphics.RenderMesh RenderMesh(this RenderGeometry renderGeometry, RenderMeshOptions renderMeshOptions = null)
+        {
+            if (renderGeometry == null)
+            {
+                BH.Engine.Base.Compute.RecordError("Cannot compute the mesh of a null render geometry object.");
```
Comment (alelom) [inline TriangleNet_Engine/Compute/Representation/RenderMesh/Other/RenderGeometry.cs]:
```suggestion
                BH.Engine.Base.Compute.RecordError($"Cannot compute the mesh of a null {nameof(RenderGeometry)} object.");
```

**Review by alelom — CHANGES_REQUESTED** [review]
Just a small change requested, code looks good and functionality works well with the provided test script.

**Comment by enarhi** [comment]
@BHoMBot check compliance

File: TriangleNet_Engine/Compute/Representation/RenderMesh/Other/RenderGeometry.cs (hunk 2 above)
Comment (enarhi) [reply TriangleNet_Engine/Compute/Representation/RenderMesh/Other/RenderGeometry.cs]: updated with improved error, thanks!

File: TriangleNet_Engine/Compute/Representation/RenderMesh/Other/RenderGeometry.cs (hunk 1 above)
Comment (enarhi) [reply TriangleNet_Engine/Compute/Representation/RenderMesh/Other/RenderGeometry.cs]: fixed

**Comment by enarhi** [comment]
@BHoMBot check compliance

**Comment by enarhi** [comment]
@BHoMBot check compliance

**Comment by enarhi** [comment]
closing, replaced by #107

**Comment by FraserGreenroyd** [comment]
Replaced by #107

**Pull request closed without merging**